src/fn/upload-custom-translations.js

```diff
@@ -81,6 +115,75 @@ function parse(filePath, options) {
   });
 }
 
+function checkTranslations(translations, lang, templatePlaceholders) {
+  let mf = null;
+  try {
+    mf = new MessageFormat(lang);
+  } catch (e) {
+    log.warn(`Cannot check '${lang}' translations: ${e.message}`);
+  }
+  const placeholders = extractPlaceholdersFromTranslations(translations);
+  let formatErrorsFound = 0;
+  let placeholderErrorsFound = 0;
+  let emptiesFound = 0;
+  for (const [msgKey, msgSrc] of Object.entries(translations)) {
+    if (!msgSrc) {
+      emptiesFound++;
+    } else if (/{{[\s\w.#^/'|]+}}/.test(msgSrc)) {
+      if (templatePlaceholders) {
+        const msgPlaceholders = placeholders[msgKey];
+        if (msgPlaceholders) {
+          const templatePlaceholder = templatePlaceholders[msgKey];
+          if (!templatePlaceholder) {
+            log.error(`Cannot compile '${lang}' translation with key '${msgKey}' ` +
+              `has placeholders, but the key does not match any of ${EN_FILE}\n` +
+              'You can use messages-ex.properties to add placeholders missing from the reference context.');
+            placeholderErrorsFound++;
+          } else {
+            const foundAllPlaceholders = msgPlaceholders.every(el => templatePlaceholder.includes(el));
+            if (!foundAllPlaceholders) {
+              log.error(`Cannot compile '${lang}' translation with key '${msgKey}' ` +
+                `has placeholders that do not match those of ${EN_FILE}\n` +
+                'You can use messages-ex.properties to add placeholders missing from the reference context.');
+              placeholderErrorsFound++;
+            }
+          }
+        }
+      }
+    } else if (mf && typeof msgSrc === 'string') {
+      try {
+        mf.compile(msgSrc);
+      } catch (e) {
+        log.error(`Cannot compile '${lang}' translation ${msgKey} = '${msgSrc}' : ${e.message}`);
+        formatErrorsFound++;
+      }
+    }
+  }
+  if (emptiesFound > 0) {
+    log.warn(transEmptyMsg({EMPTIES: emptiesFound}) + ` '${lang}' translations`);
+  }
+  if (formatErrorsFound > 0 || placeholderErrorsFound > 0) {
+    log.error(transErrorsMsg({ERRORS: formatErrorsFound + placeholderErrorsFound})
+              + ` '${lang}' translations`);
+    process.exit(-1);
+  }
+}
+
+function extractPlaceholdersFromTranslations(translations, extraPlaceholders = {}) {
+  // Extract from github.com/medic/cht-core/blob/master/scripts/poe/lib/utils.js
+  const result = {};
+  for (const [msgKey, msgSrc] of Object.entries(translations)) {
+    let placeholders = typeof msgSrc === 'string' ? msgSrc.match(/{{[\s\w.#^/'|]+}}/g) : null;
+    if (placeholders) {
+      const msgExtraPlaceholders = extraPlaceholders[msgKey] ? extraPlaceholders[msgKey] : [];
+      result[msgKey] = _.uniq(placeholders.concat(msgExtraPlaceholders));
```

Comment:
I think now that you're using lodash this can be made simpler...
```suggestion
      result[msgKey] = _.uniq(placeholders.concat(extraPlaceholders[msgKey] || []));
```

What do you think?